esphome/components/mqtt/mqtt_climate.cpp

```diff
@@ -66,12 +66,36 @@ void MQTTClimateComponent::send_discovery(JsonObject root, mqtt::SendDiscoveryCo
   // temperature units are always coerced to Celsius internally
   root[MQTT_TEMPERATURE_UNIT] = "C";
 
-  if (traits.supports_preset(CLIMATE_PRESET_AWAY)) {
-    // away_mode_command_topic
-    root[MQTT_AWAY_MODE_COMMAND_TOPIC] = this->get_away_command_topic();
-    // away_mode_state_topic
-    root[MQTT_AWAY_MODE_STATE_TOPIC] = this->get_away_state_topic();
+  if (traits.get_supports_presets() || !traits.get_supported_custom_presets().empty()) {
```

Comment:
You added this check inside `get_supports_presets()`
```suggestion
  if (traits.get_supports_presets()) {
```